src/threshold_sign.rs

```diff
@@ -135,19 +153,43 @@ impl<N: NodeIdT> ThresholdSign<N> {
             return Ok(Step::default());
         }
         let Message(share) = message;
-        if let Some(pk_i) = self.netinfo.public_key_share(sender_id) {
-            if !pk_i.verify_g2(&share, self.msg_hash) {
-                // Log the faulty node and ignore the invalid share.
-                let fault_kind = FaultKind::UnverifiedSignatureShareSender;
-                return Ok(Fault::new(sender_id.clone(), fault_kind).into());
-            }
-            self.received_shares.insert(sender_id.clone(), share);
+        if !self.is_share_valid(sender_id, &share) {
+            let fault_kind = FaultKind::UnverifiedSignatureShareSender;
+            return Ok(Fault::new(sender_id.clone(), fault_kind).into());
```

Comment:
I think we should keep the old behavior here: If the sender is unknown, return an error.
Actually, we should change `ThresholdDecryption` to do the same.